app/src/main/java/org/mozilla/tv/firefox/FirefoxApplication.kt

```diff
@@ -70,6 +74,24 @@ open class FirefoxApplication : LocaleAwareApplication() {
                 updateExperiments()
             }
 
+            // This is needed for Rust components to make HTTP network requests
+            Megazord.init()
+            RustHttpConfig.setClient(lazy { HttpUrlConnectionWrapper.client })
+
+            var admAvailable = false
+            try {
+                Class.forName("com.amazon.device.messaging.ADM")
```

Comment:
I'm not sure how much I trust the Amazon [guide re graceful degradation](https://developer.amazon.com/docs/adm/integrate-your-app.html#gracefully-degrade-if-adm-is-unavailable). This check should technically pass since we are bundling the ADM jar that contains this class even if the device is not supported. 

It seems like there is also a [`ADM.isSupported`](https://s3-us-west-1.amazonaws.com/devportal-reference-docs/adm/API-Reference/reference/com/amazon/device/messaging/ADM.html#isSupported()) as well that might be an alternative or can be a secondary check in this block as well:

```kotlin
admAvailable = ADM.isSupported()
```

Reply:
> This check should technically pass since we are bundling the ADM jar that contains this class even if the device is not supported.

The ADM jar is compileOnly and the classes are provided on device so I assume their check would work.

That being said, `ADM.isSupported()` sounds much better if it's available on all devices.

Reply:
> The ADM jar is compileOnly and the classes are provided on device so I assume their check would work.

Nice, I didn't notice that fire TV needed to do that as well.